components/net/indexeddb/engines/sqlite.rs

```diff
@@ -0,0 +1,513 @@
+                None => {
+                    return Err(Self::Error::Custom("No such store".to_string()));
+                },
+            };
+            if object_store_index_model::Entity::find()
+                .filter(
+                    object_store_index_model::Column::Name
+                        .eq(index_name.to_string())
+                        .and(object_store_index_model::Column::ObjectStoreId.eq(object_store.id)),
+                )
+                .one(&self.connection)
+                .await?
+                .is_some()
+            {
+                return Ok(CreateObjectResult::AlreadyExists);
+            }
+            let model = object_store_index_model::ActiveModel {
+                id: Default::default(),
+                object_store_id: Set(object_store.id),
+                name: Set(index_name),
+                key_path: Set(bincode::serialize(&key_path).unwrap()),
+                unique_index: Set(unique),
+                multi_entry_index: Set(multi_entry),
+            };
+            model.insert(&self.connection).await?;
+            Ok(CreateObjectResult::Created)
+        })
+    }
+
+    fn delete_index(
+        &self,
+        store_name: SanitizedName,
+        index_name: String,
+    ) -> Result<(), Self::Error> {
+        HANDLE.block_on(async {
+            let object_store = match object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await?
+            {
+                Some(model) => model,
+                None => {
+                    return Err(Self::Error::Custom("No such store".to_string()));
+                },
+            };
+            if let Some(model) = object_store_index_model::Entity::find()
+                .filter(
+                    object_store_index_model::Column::Name
+                        .eq(index_name.to_string())
+                        .and(object_store_index_model::Column::ObjectStoreId.eq(object_store.id)),
+                )
+                .one(&self.connection)
+                .await?
+            {
+                model.delete(&self.connection).await?;
+            }
+            Ok(())
+        })
+    }
+
+    fn version(&self) -> u64 {
+        HANDLE.block_on(async {
+            let db_info = database_model::Entity::find()
+                .one(&self.connection)
+                .await
+                .unwrap()
+                .unwrap();
+            db_info.version
+        }) as u64
+    }
+
+    fn set_version(&self, version: u64) -> Result<(), Self::Error> {
+        HANDLE.block_on(async {
+            let db_info = database_model::Entity::find()
+                .one(&self.connection)
+                .await
+                .unwrap()
+                .unwrap();
+            let mut db_info = db_info.into_active_model();
+            db_info.version = Set(version as i32);
```

Comment:
I'm nervous that our public API takes values that are unrepresentable in the actual implementation. What can we do about that?

Reply:
Good point, maybe just using an i64 should solve the issue.